src/coreclr/vm/invokeutil.cpp

```diff
@@ -124,191 +124,115 @@ void *InvokeUtil::GetIntPtrValue(OBJECTREF pObj) {
     RETURN *(void **)((pObj)->UnBox());
 }
 
-void InvokeUtil::CopyArg(TypeHandle th, OBJECTREF *pObjUNSAFE, ArgDestination *argDest) {
+void InvokeUtil::CopyArg(TypeHandle th, PVOID **pArgRef, ArgDestination *argDest) {
     CONTRACTL {
         THROWS;
         GC_NOTRIGGER; // Caller does not protect object references
         MODE_COOPERATIVE;
         PRECONDITION(!th.IsNull());
-        PRECONDITION(CheckPointer(pObjUNSAFE));
         INJECT_FAULT(COMPlusThrowOM());
     }
     CONTRACTL_END;
 
     void *pArgDst = argDest->GetDestinationAddress();
+    PVOID *rArg = *pArgRef;
+    CorElementType type = th.GetVerifierCorElementType();
 
-    OBJECTREF rObj = *pObjUNSAFE;
-    MethodTable* pMT;
-    CorElementType oType;
-    CorElementType type;
-
-    if (rObj != 0) {
-        pMT = rObj->GetMethodTable();
-        oType = pMT->GetInternalCorElementType();
-    }
-    else {
-        pMT = 0;
-        oType = ELEMENT_TYPE_OBJECT;
-    }
-    type = th.GetVerifierCorElementType();
-
-    // This basically maps the Signature type our type and calls the CreatePrimitiveValue
-    //  method.  We can omit this if we get alignment on these types.
     switch (type) {
     case ELEMENT_TYPE_BOOLEAN:
-    case ELEMENT_TYPE_I1:
+    case ELEMENT_TYPE_CHAR:
     case ELEMENT_TYPE_U1:
+    case ELEMENT_TYPE_I1:
+        _ASSERTE(rArg != NULL);
+        *(PVOID *)pArgDst = (INT8*)*rArg;
```

Comment:
These casts are very unusual. They are relying upon the degrading of pointers to `void*`. I think I'd match both sides as opposed what is going on here.

Reply:
Although I'm not sure, I believe `PVOID` was used here originally because the underlying destination is a 32- or 64-bit pointer. Otherwise it would be INT32 or something equivalent.

Reply:
The `PVOID`s were removed in a later commit.